articles/how-to-build-a-todo-app/index.md

```diff
@@ -0,0 +1,216 @@
+
+![Flutter Default App](/engineering-education/how-to-build-a-todo-app/default.jpg)
+
+**Flutter default after running any app for the first time**
+
+Go to the `main.dart` file which is in the lib directory. Change the default code to the snippet shown in the below picture.
+
+![Stateless Widget](/engineering-education/how-to-build-a-todo-app/stateless.jpg)
+
+After running the app you will notice an empty canvas with the title To-do-List. Let us look at the code.
+
+![Stateless Widget Code](/engineering-education/how-to-build-a-todo-app/stateless.jpg)
+
+### Flutter Packages
+On the first line, we imported a flutter package named `Material.dart` (import 'package:flutter/material.dart';) to make app development fast flutter comes with a package that makes it easy to start building a material style app. Packages are simply other people's solution for a feature you need in your app. Instead of building some features from scratch, you could just go to [pub.dartlang.org](pub.dartlang.org) and search for a package that will perform that feature you want to build. You can incorporate it into your application. This will save a lot of time for you as a developer. As you develop with flutter, you will use a lot of packages.
+
+### Flutter Widgets
+Flutter Widgets describe what the view of an app should look like given the current configuration and state. According to the Flutter documentation, Flutter widgets are built using a modern framework that takes inspiration from React.
+
+A widget can help with layout, define design, etc. For example: Padding, Margin, Center, Layout rows, and columns are all widgets.
+
+From our code, the whole app is a widget which contains a MaterialApp widget.
+- The scaffold is the widget that helps us create a proper material layout without the worry of manual styling.
+- The AppBar is a widget that accepts a title and creates a bar at the top of the screen, this is normal in apps. This aligns the text to the left on Android and aligns text to the center on iOS. 
+
+### Stateless Widget
+A stateless widget is a widget whose state cannot be changed once it is built. i.e. no amount of change in the variables, icons, buttons, or retrieving data can change the state of the app.
+A to-do app will always have to-do items added and remove, to achieve this we will need to implement a stateful widget.
+
+### Stateful Widget
+This type of widget is dynamic. This means it can change its appearance when it receives data or change appearance in response to events triggered.
+
+Edit your code to match the one in the picture below.
+
+![Stateful Widget](/engineering-education/how-to-build-a-todo-app/stateful.jpg)
```

Comment:
You've gotta add your code as code, and not as an image. Please delete this image and add the code within code blocks.